src/jit/optimizer.cpp

```diff
@@ -4648,7 +4648,8 @@ void Compiler::optCloneLoop(unsigned loopInd, LoopCloneContext* context)
         BasicBlock* newBlk = fgNewBBafter(blk->bbJumpKind, newPred,
                                           /*extendRegion*/ true);
 
-        BasicBlock::CloneBlockState(this, newBlk, blk);
+        bool cloneOk = BasicBlock::CloneBlockState(this, newBlk, blk);
+        noway_assert(cloneOk);
```

Comment:
That would be good.  It wasn't clear to me how the invariant that success would be returned by `BasicBlock::CloneBlockState `was being enforced.